### DISCUSSION TITLE
How to configure external Diff tool?

### DISCUSSION THREAD

**Discussion by mbnoimi** [jorio/gitfourchette#4] [Q&A]
I use flatpak version of `meld`. How can I configure Gitfourchette to accept it?

The following failed!

```
/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=meld --file-forwarding org.gnome.meld $L $R
/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=meld --file-forwarding org.gnome.meld --auto-merge $L $B $R --output=$M
```

![image](https://github.com/user-attachments/assets/53b6a863-3a57-4dcf-8515-9be36dcdca4e)

**Comment by mbnoimi** [comment]
More screenshots:

![image](https://github.com/user-attachments/assets/67331278-03ff-4afd-8be4-4ab1870c75bd)

![image](https://github.com/user-attachments/assets/cd9992cb-6b14-4983-ae3c-f50d269f7080)

```ini
$ flatpak info --show-permissions org.gitfourchette.gitfourchette
[Context]
shared=network;ipc;
sockets=x11;wayland;fallback-x11;
filesystems=xdg-config/kdeglobals:ro;/usr/share/themes;host;/home;

[Session Bus Policy]
com.canonical.AppMenu.Registrar=talk
org.kde.kconfig.notify=talk
org.kde.KGlobalSettings=talk
org.freedesktop.Flatpak=talk
org.freedesktop.FileManager1=talk

[Environment]
GTK_THEME=Mint-Y-Dark-Aqua
QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess
PYTHONPYCACHEPREFIX=
QT_STYLE_OVERRIDE=Adwaita-dark

$ flatpak info --show-permissions org.gnome.meld
[Context]
shared=ipc;
sockets=x11;wayland;fallback-x11;
devices=dri;
filesystems=/usr/share/themes;host;/home;

[Environment]
GTK_THEME=Mint-Y-Dark-Aqua
QT_STYLE_OVERRIDE=Adwaita-dark


```

**Comment by jorio** [comment]
Right now GitFourchette writes temporary files to drive external merge tools. In the Flatpak version, that's in a sandboxed directory, and indeed Flatpaks cannot access each others' sandboxes. I'll investigate a fix. Thank you for reporting this!

**Comment by jorio** [comment]
The workaround below isn't elegant, but this should let you use Meld while I cook up a proper fix.

Diff tool:
```
flatpak run --filesystem=/run/user/1000/app/org.gitfourchette.gitfourchette org.gnome.meld $L $R
```

Merge tool:
```
flatpak run --filesystem=/run/user/1000/app/org.gitfourchette.gitfourchette org.gnome.meld --auto-merge $L $B $R --output=$M
```

**Reply by mbnoimi** [reply]
Thanks. I will use it for now and I wish to release a practical fix soon.
